portal-kernel/src/com/liferay/portal/kernel/servlet/BrowserSnifferUtil.java

```diff
@@ -15,120 +23,443 @@
  */
 public class BrowserSnifferUtil {
 
+	public static final String BROWSER_ID_CHROME = "chrome";
+
+	public static final String BROWSER_ID_EDGE = "edge";
+
+	public static final String BROWSER_ID_FIREFOX = "firefox";
+
+	public static final String BROWSER_ID_IE = "ie";
+
+	public static final String BROWSER_ID_OTHER = "other";
+
+	public static final String[] REVISION_LEADINGS = {
```

Comment:
@julschong no reason to expose these constants array as public. They are only ever referenced by tests, keep them private, tests can use reflection to get those fields.